File: src/main/java/no/nav/bidrag/beregn/saertilskudd/beregning/SaertilskuddBeregning.java
package no.nav.bidrag.beregn.saertilskudd.beregning;

import java.util.List;
import no.nav.bidrag.beregn.saertilskudd.bo.GrunnlagBeregning;
import no.nav.bidrag.beregn.saertilskudd.bo.ResultatBeregning;

public interface SaertilskuddBeregning {

  ResultatBeregning beregn(
      GrunnlagBeregning grunnlagBeregning);

  static SaertilskuddBeregning getInstance() {
    return new SaertilskuddBeregningImpl();
  }

}

File: src/main/kotlin/no/nav/bidrag/beregn/saertilskudd/beregning/SaertilskuddBeregning.kt
package no.nav.bidrag.beregn.saertilskudd.beregning

import no.nav.bidrag.beregn.saertilskudd.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.saertilskudd.bo.ResultatBeregning

fun interface SaertilskuddBeregning {

    fun beregn(grunnlag: GrunnlagBeregning): ResultatBeregning

    companion object {
        fun getInstance(): SaertilskuddBeregning = SaertilskuddBeregningImpl()
    }
}
